cmd/ntpcheck/cmd/utils.go

```diff
@@ -108,13 +124,29 @@ func ntpDate(remoteServerAddr string, remoteServerPort string, requests int) err
 		return err
 	}
 
-	fmt.Printf("Server: %s, Requests: %d\n", addr, requests)
-	var sumAvgNetworkDelay int64
+	var sumDelay int64
 	var sumOffset int64
+	var clientTransmitTime time.Time
+	var actualTransmitTime time.Time
 
 	for i := 0; i < requests; i++ {
-		clientTransmitTime := time.Now()
+		actualTransmitTime = time.Now()
+		if clientTransmitOffset != 0 {
+			clientTransmitTime = actualTransmitTime.Add(clientTransmitOffset)
+		} else if userDate != "" {
+			clientTransmitTime, err = dateparse.ParseLocal(userDate)
```

Comment:
This sort of check (user input verification) should probably happen before we even start dialing. Just like you did for `ntpdateTransmitOffset != 0 && ntpdateUserDate != ""` checks. If there is no ntpdateUserDate specified by user - let's pass in `time.Time{}` and check for `userDate.IsZero()` instead of `userDate != ""`